bundles/plugins/org.bonitasoft.studio.contract/src/org/bonitasoft/studio/contract/core/mapping/SimpleFieldToContractInputMapping.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (C) 2015 Bonitasoft S.A.
+ * Bonitasoft, 32 rue Gustave Eiffel - 38000 Grenoble
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2.0 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.bonitasoft.studio.contract.core.mapping;
+
+import org.bonitasoft.engine.bdm.model.field.FieldType;
+import org.bonitasoft.engine.bdm.model.field.SimpleField;
+import org.bonitasoft.studio.model.process.ContractInputType;
+
+/**
+ * @author aurelie
+ */
+public class SimpleFieldToContractInputMapping extends FieldToContractInputMapping {
+
+    private final SimpleField simpleField;
+
+    public SimpleFieldToContractInputMapping(final SimpleField field) {
+        super(field);
+        simpleField = field;
+    }
+
+    @Override
+    protected ContractInputType toContractInputType() {
+        final FieldType fieldType = simpleField.getType();
+        switch (fieldType) {
+            case STRING:
+                return ContractInputType.TEXT;
+            case TEXT:
+                return ContractInputType.TEXT;
+            case INTEGER:
+                return ContractInputType.INTEGER;
+            case LONG:
+                return ContractInputType.INTEGER;
+            case DOUBLE:
+                return ContractInputType.DECIMAL;
+            case FLOAT:
+                return ContractInputType.DECIMAL;
+            case BOOLEAN:
+                return ContractInputType.BOOLEAN;
+            case DATE:
+                return ContractInputType.DATE;
+            default:
+                throw new IllegalStateException(String.format("Failed to convert field type %s to contract input type", fieldType));
```

Comment:
Hell no. We handle all cases used in the studio (there is more FieldType available than the one we uses in the studio.)